commit 65da3773e67361d03a210a83e567482d0c1f8070 (refs/changes/34/3634/3)
Author: Patrick Mooney <pmooney@pfmooney.com>
Date:   2018-03-14T23:28:34+00:00 (1 year, 7 months ago)
    
    OS-6711 ship bhyve bootrom binary
    Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com>
    Approved by: Jerry Jelinek <jerry.jelinek@joyent.com>

diff --git a/overlay/generic/manifest b/overlay/generic/manifest
index 8683f59c..64d0f21b 100644
--- a/overlay/generic/manifest
+++ b/overlay/generic/manifest
@@ -174,6 +174,8 @@ f usr/lib/localedef/extensions/UTF-8.x 0444 root bin
 f usr/sbin/fssstat 0555 root bin
 f usr/sbin/zonememstat 0555 root bin
 f usr/sbin/zonemon 0555 root bin
+d usr/share/bhyve 0555 root bin
+f usr/share/bhyve/uefi-csm-rom.bin 0444 root bin
 f usr/share/man/man1/json.1 0444 root bin
 f usr/share/man/man1m/fssstat.1m 0444 root bin
 d var/db 0755 root sys
diff --git a/overlay/generic/usr/share/bhyve/README b/overlay/generic/usr/share/bhyve/README
new file mode 100644
index 00000000..a1bb7a08
--- /dev/null
+++ b/overlay/generic/usr/share/bhyve/README
@@ -0,0 +1,6 @@
+The uefi-csm-rom.bin file is a binary generated from the uefi-edk2 package
+shipped in illumos-extra.  At the time of this writing, facilitating an
+isolated build of the uefi ROM was difficult thanks to its dependency on iasl
+to compile the ACPI tables it ships.  While it is excluded from the build until
+those issues are solved, the ROM can be built from those sources (and patches)
+provided a working iasl can be configured for the task.
diff --git a/overlay/generic/usr/share/bhyve/uefi-csm-rom.bin b/overlay/generic/usr/share/bhyve/uefi-csm-rom.bin
new file mode 100755
index 00000000..02f11ea5
Binary files /dev/null and b/overlay/generic/usr/share/bhyve/uefi-csm-rom.bin differ
